crates/services/consensus_module/poa/src/pre_confirmation_signature_service.rs

```diff
@@ -219,3 +226,60 @@ where
         Ok(())
     }
 }
+
+pub type Service<TxRcv, Brdcst, Parent, Gen, DelegateKey, Trigger> = ServiceRunner<
+    PreConfirmationSignatureTask<TxRcv, Brdcst, Parent, Gen, DelegateKey, Trigger>,
+>;
+
+pub fn new_service<TxRcv, Brdcst, Parent, Gen, DelegateKey, Trigger, Preconfirmations>(
+    config: config::Config,
+    preconfirmation_receiver: TxRcv,
+    mut p2p_adapter: Brdcst,
+    parent_signature: Arc<Parent>,
+    mut key_generator: Gen,
+    key_rotation_trigger: Trigger,
+) -> anyhow::Result<Service<TxRcv, Brdcst, Parent, Gen, DelegateKey, Trigger>>
+where
+    TxRcv: TxReceiver<Txs = Preconfirmations>,
+    Brdcst: Broadcast<
+        DelegateKey = DelegateKey,
+        ParentKey = Parent,
+        Preconfirmations = Preconfirmations,
+    >,
+    Gen: KeyGenerator<Key = DelegateKey>,
+    Trigger: KeyRotationTrigger,
+    DelegateKey: SigningKey,
+    Parent: ParentSignature,
+    Preconfirmations: serde::Serialize + Send + Debug,
+{
+    // It's ok to wait because the first rotation is expected to be immediate
+    let expiration = Tai64(
+        Tai64::now()
+            .0
+            .saturating_add(config.key_expiration_interval.as_secs()),
+    );
+    let (new_delegate_key, sealed) = futures::executor::block_on(create_delegate_key(
+        &mut key_generator,
+        &parent_signature,
+        expiration,
+    ))
+    .map_err(|e| anyhow::anyhow!(e))?;
+
+    if let Err(e) = futures::executor::block_on(
```

Comment:
We shouldn't do `block_on` in the constructor. It is why `into_task` exists, to init the service, if you need to. Like create a first delegation for example and broadcast it=)

Reply:
Done.